lms/djangoapps/discussion/rest_api/utils.py

```diff
@@ -33,13 +36,15 @@ class AttributeDict(dict):
 
 def discussion_open_for_user(course, user):
     """
-    Check if course discussion are open or not for user.
+    Check if the course discussion are open or not for user.
 
     Arguments:
             course: Course to check discussions for
             user: User to check for privileges in course
     """
-    return course.forum_posts_allowed or has_discussion_privileges(user, course.id)
+    discussions_config = DiscussionsConfiguration.get(course.id).posting_restrictions
```

Comment:
Can we rename this to something more specific. Like `discussions_posting_restrictions`